### PR TITLE
How to guide on Dask-Nebari config 

### LINKED ISSUE #68: [DOC] How To configure Dask on Nebari
How do I configure dask on nebari?

This is a doc that administrators will be reading. 

* what do I need to include in the nebari config?
* what are the related infra parts?
  * Dask-gateway pod is like a manager for any job request that a user makes and it comes witht he nebari deployment (static) toghther with the dask-controller;
  * Dask-scheduler is created open request from a user, and its managed by dask-gateway
  * Dask worker is the actual place where the computation occur, for each job you can request # of workers that will run the job in parallel
* note that to use it, the env needs to include dask (or qhub-dask?)
* what is qhub-dask?

### PR DESCRIPTION
**Pull request by sayantikabanik**
resolves #68 

This PR is a work in progress. I am currently referring to [this](https://docs.qhub.dev/en/latest/source/installation/configuration.html?highlight=dask#full-configuration-example) guide to pick bits and pieces.

### CODE REVIEW

**Comment by sayantikabanik** [comment]
@HarshCasper @iameskild feel free to continue on this PR or close it.

**Comment by iameskild** [comment]
@HarshCasper could you look at [integrating these docs](https://docs.qhub.dev/en/latest/source/installation/configuration.html?highlight=dask#dask-profiles) into this how to guide?

File: docs/docs/how-tos/dask-nebari-config.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+---
+id: dask-nebari-config
+title: Configure Dask on Nebari
+---
+
+# Introduction
+
+In this tutorial we will dive into the `Nebari-Dask` configuration details. Nebari config is essentially
+a `yaml` file which is at the heart of all things (most of them) related to configurations.
+Our main focus in this tutorial will be the `profiles` & `dask_worker` section of the config file.
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:10]:
```suggestion
In this tutorial we will dive into the configuration requirements for running Daask on Nebari. The Nebari config (`qhub_config.yml`) is at the heart of all things (most of them) related to configurations.
Our main focus in this tutorial will be the `profiles` & `dask_worker` section of the config file.
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+### Basic infrastructure details
+
+Before we dive deeper in configuration details, let's understand about how are the core configuration
+components.
+
+### Core components:
+
+- Dask-gateway
+- dask workers
+- Dask scheduler
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:21]:
```suggestion
There are three core configuration components necessary for setting up Dask on Nebari.

These are:
- [Dask-gateway](https://gateway.dask.org/): provides a secure, multi-tenant server for managing [Dask](https://dask.org/) clusters
- [Dask workers](https://distributed.dask.org/en/stable/worker.html): compute tasks as directed by the scheduler and store/serve results
- [Dask scheduler](https://docs.dask.org/en/stable/scheduler-overview.html): executes the task graph by coordinating task distribution
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+title: Configure Dask on Nebari
+---
+
+# Introduction
+
+In this tutorial we will dive into the `Nebari-Dask` configuration details. Nebari config is essentially
+a `yaml` file which is at the heart of all things (most of them) related to configurations.
+Our main focus in this tutorial will be the `profiles` & `dask_worker` section of the config file.
+
+### Basic infrastructure details
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:12]:
```suggestion
## Basic infrastructure details
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+Before we dive deeper in configuration details, let's understand about how are the core configuration
+components.
+
+### Core components:
+
+- Dask-gateway
+- dask workers
+- Dask scheduler
+
+### How to configure dask gateway profiles?
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:23]:
```suggestion
## Configuring Dask Gateway profiles
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    "Huge Worker":
+      worker_cores_limit: 7
+      worker_cores: 6
+      worker_memory_limit: 30G
+      worker_memory: 28G
+      scheduler_cores_limit: 7
+      scheduler_cores: 6
+      scheduler_memory_limit: 30G
+      scheduler_memory: 28G
+```
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:82]:
```suggestion
```

## Next Steps

Now that you have your Nebari instance all set up to run Dask, check out our user guide on [Working with Big Data using Dask](/tutorials/using_dask)!

```

Comment (kcpevey) [reply docs/docs/how-tos/dask-nebari-config.md:82]: Just guessing at that link to the page, you should double check that works

File: docs/docs/how-tos/dask-nebari-config.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+### How to configure dask gateway profiles?
+
+Nebari allows for the configuration of the Dask workers. Similar to the JupyterLab instances, you only need to configure the cores and memory.
+
+When configuring the memory and CPUs for profiles, some important considerations exist. Two essential terms to understand are:
+
+-  `limit` is the absolute max memory a given pod can consume. Suppose a process within the pod consumes more than the  `limit`  memory. In that case, the Linux OS will kill the process. `limit` is not used for scheduling purposes with Kubernetes.
+    
+-  `guarantee`: is the amount of memory the Kubernetes scheduler uses to place a given pod. In general, the `guarantee`  will be less than the limit. Often the node itself has less available memory than the node specification. See this  [guide from digital ocean](https://docs.digitalocean.com/products/kubernetes/#allocatable-memory), which generally applies to other clouds.
+
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:32]:
```suggestion

The following is an example configuration:
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+### How to configure dask gateway profiles?
+
+Nebari allows for the configuration of the Dask workers. Similar to the JupyterLab instances, you only need to configure the cores and memory.
+
+When configuring the memory and CPUs for profiles, some important considerations exist. Two essential terms to understand are:
+
+-  `limit` is the absolute max memory a given pod can consume. Suppose a process within the pod consumes more than the  `limit`  memory. In that case, the Linux OS will kill the process. `limit` is not used for scheduling purposes with Kubernetes.
+    
+-  `guarantee`: is the amount of memory the Kubernetes scheduler uses to place a given pod. In general, the `guarantee`  will be less than the limit. Often the node itself has less available memory than the node specification. See this  [guide from digital ocean](https://docs.digitalocean.com/products/kubernetes/#allocatable-memory), which generally applies to other clouds.
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:31]:
```suggestion
-  `guarantee`: is the amount of memory the Kubernetes scheduler uses to place a given pod. In general, the `guarantee`  will be less than the limit. Often the node itself has less available memory than the node specification. You may want to check out this [guide from digital ocean](https://docs.digitalocean.com/products/kubernetes/#allocatable-memory) which also generally applies to other clouds.
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+      description: Stable environment with 2 cpu / 4 GB ram
+      access: keycloak
+      kubespawner_override:
+        cpu_limit: 2
+        cpu_guarantee: 2
+        mem_limit: 4G
+        mem_guarantee: 4G
+```
+
+### How to configure dask scheduler?
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:67]:
```suggestion
## Configuring the Dask Scheduler
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+In a few instances, the dask worker node-group might be running on quite a large instance, perhaps with 8 CPUs and 32 GB of memory (or more). In this case, you might also want to increase the resource levels associated with the dask scheduler.
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:69]:
```suggestion
For analyses requiring heavy compute, there may be some situations where the Dask worker node-group might be running on quite a large cluster, perhaps with 8 CPUs and 32 GB of memory (or more). In this case, you might also want to increase the resource levels associated with the Dask scheduler.
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:70]:
```suggestion

The following is an example of a Dask worker configuration with the Scheduler resources modified.
```

File: docs/docs/how-tos/dask-nebari-config.md [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+dask_worker:
+    "Huge Worker":
+      worker_cores_limit: 7
+      worker_cores: 6
+      worker_memory_limit: 30G
+      worker_memory: 28G
+      scheduler_cores_limit: 7
+      scheduler_cores: 6
+      scheduler_memory_limit: 30G
+      scheduler_memory: 28G
```
Comment (kcpevey) [inline docs/docs/how-tos/dask-nebari-config.md:81]: Is this a jupyterlab profile?

**Review by kcpevey — CHANGES_REQUESTED** [review]

**Comment by pavithraes** [comment]
The content in this PR is covered in #115. :)

**Pull request closed without merging**